src/clj/puppetlabs/services/jruby/jruby_puppet_schemas.clj

```diff
@@ -149,25 +99,20 @@
                      (nil? (schema/check JRubyInstanceState @%)))
                'JRubyInstanceState))
 
-;; A record representing an individual entry in the JRubyPuppet pool.
+;; TODO: rename, get rid of references to puppet
 (schema/defrecord JRubyPuppetInstance
-                  [pool :- pool-queue-type
-                   id :- schema/Int
-                   max-requests :- schema/Int
-                   flush-instance-fn :- IFn
-                   state :- JRubyInstanceStateContainer
-                   jruby-puppet :- JRubyPuppet
-                   scripting-container :- ScriptingContainer
-                   environment-registry :- (schema/both
```

Comment:
Yeah, something like that... or like a callback that allows the user to modify the object after initialization, and return a modified copy that we will use in the pool.  In that case it could maybe be a subclass or something, and then they could decide how they want to handle the additional state on their own.